components/tidb_query/src/codec/mysql/json/mod.rs

```diff
@@ -112,6 +112,27 @@ impl ConvertTo<Decimal> for Json {
     }
 }
 
+impl ConvertTo<Json> for i64 {
+    #[inline]
+    fn convert(&self, _: &mut EvalContext) -> Result<Json> {
+        Ok(Json::I64(*self))
```

Comment:
```go
func (b *builtinCastIntAsJSONSig) evalJSON(row chunk.Row) (res json.BinaryJSON, isNull bool, err error) {
	val, isNull, err := b.args[0].EvalInt(b.ctx, row)
	if isNull || err != nil {
		return res, isNull, err
	}
	if mysql.HasIsBooleanFlag(b.args[0].GetType().Flag) {
		res = json.CreateBinary(val != 0)
	} else if mysql.HasUnsignedFlag(b.args[0].GetType().Flag) {
		res = json.CreateBinary(uint64(val))
	} else {
		res = json.CreateBinary(val)
	}
	return res, false, nil
}
```

Do we need to impl the logic above here?

Reply:
There is no need to handle this case due to:
1. there is no `FieldType` to indicate whether `IS_BOOLEAN` flag is set
2. the `impl_cast.rs` doesn't use this to implement CAST int as JSON.